phpBB/phpbb/textformatter/s9e/utils.php

```diff
@@ -35,6 +35,43 @@ public function clean_formatting($xml)
 	}
 
 	/**
+	* Return given string between quotes
+	*
+	* Will use either single- or double- quotes depending on whichever requires to be escaped.
+	* Quotes and backslashes are escaped with backslashes where necessary
+	*
+	* @param  string $str Original string
+	* @return string      Escaped string within quotes
+	*/
+	protected function enquote($str)
+	{
+		$quote = (strpos($str, '"') === false || strpos($str, "'") !== false) ? '"' : "'";
```

Comment:
It uses singles quotes if the string contains a double quote and no single quotes, otherwise it uses double quotes. However, I'm going to change that in a second that that it uses whichever form is the shortest. Things that need to be escaped are always escaped regardless.